src/derive/curve.rs

```diff
@@ -231,61 +372,93 @@ macro_rules! new_curve_impl {
                         }
 
                         fn from_uncompressed_unchecked(bytes: &Self::Uncompressed) -> CtOption<Self> {
-                            let bytes = &bytes.0;
-                            let infinity_flag_set = Choice::from((bytes[[< $name _UNCOMPRESSED_SIZE >] - 1] >> 6) & 1);
-                            // Attempt to obtain the x-coordinate
+                            let mut bytes = bytes.0;
+
+                            let flag_idx = 2* $base::size() -1;
+
+                            // In the uncompressed format, the bit in the sign flag position must be 0 always.
+                            let sign_flag =  if $spare_bits == 2 || $spare_bits == 1 {
+                                Choice::from( (bytes[ flag_idx ] & SIGN_MASK) >> SIGN_SHIFT )
+                            } else {
+                                // For 0 spare bits, there is no sign flag.
+                                Choice::from(0u8)
+                            };
+
+                            // Get identity flag.
+                            let identity_flag= if $spare_bits == 2 {
+                                let identity_flag = Choice::from( ( ( bytes[ flag_idx ] & IDENTITY_MASK) >> IDENTITY_SHIFT) );
```

Comment:
@davidnevadoc 
This is where I mentioned about `mismatch`.
The `flag_idx` is the `2 * $base::size() - 1`.
Hence, the flag byte would be the MSB of whole encoded bytes.
And, the order of encoding/decoding coordinates is `y || x`, according to proposal & doc comments.
It means that we get the flag byte from the MSB of `x` coordinate bytes.
Then, it is opposite of the proposal for `2 spare bits uncompressed format`.
```
| ------ | ----  | ------------ | ------ | ---- | ------------ |
| 0      | ident | y-coordinate | 0      | 0    | x-coordinate |
| ------ | ----  | ------------ | ------ | ---- | ------------ |
  1 bit    1 bit                  1 bit    1 bit
  ```
  Is it what you want?

Reply:
I have made a proposal in this commit https://github.com/privacy-scaling-explorations/halo2curves/commit/855ddfb43d550668af4c5cf2f58a5c1ef389675f please take a look!
In the proposal I did:
- In the tables add 3 more rows that indicate: byte position in little endian, byte position in big endian, bit position (within a byte)
- Reword the description to be general for little endian / big endian

I think that with this changes it becomes very clear what's the encoding with little endian and big endian, and hopefully there's no room for doubts, as it's very explicit in all details.  Let me know what you think!

Reply:
Thanks! @ed255 

One question(for clarity)
What is custom/norm when indicating byte positions of the LE/BE?

For example, it shows that byte positions are ordered from `BS - 1` to `0` for `LE` in the `1 spare bit compressed format` diagram. And vice versa in `BE`.
```
        // According to the number of spare bits.
        // 1 Spare bit:
        //
        //     |                  | sign      | x-coordinate |
        //     | Byte pos. (LE)   | BS-1          ..       0 |
        //     | Byte pos. (BE)   | 0          ..       BS-1 |
        //     | Bit pos.         | 7         |              |
        //     | ---------------- | --------  | ------------ |
        //     | Identity         | 0         | 0            |
        //     | Non-identity $P$ | $sgn0(P)$ | $P.x$        |
```
So far, I believe that the byte positions in `LE` is ordered from `0` to `BS - 1`.
In other words, the first byte(0) of bytes is the LSB of encoded element, and last byte(BS - 1) is the MSB in `LE`.
Then, the diagram shows the opposite ordering.
So, the `Byte pos.` is not the index of individual byte in bytes array?

Reply:
Hmm, maybe, I am complicating the things. 💦 
I understand that the `FLAG_BYTE(sign, identity)` is the `BS - 1` byte/MSB byte of the x/y coordinate.

Reply:
Here's my understanding:
- Endianess only refers to encoding integers into byte arrays
- Little Endian encodes the least significant bytes in the beginning of the array and the most significant ones at the end
- Big Endian encodes the most significant bytes in the beginning of the array and the least significant ones at the end.
- The least significant byte of an integer encoding is the byte that contributes to the smallest value of the integer
- The most significant byte of an integer encoding is the byte that contributes the biggest value of the integer.
- For example let's take the 16 bit number `500 = 232 + 3 * 2 ^ 8`
  - 232 is the least significant byte
  - 3 is the most significant byte
  - little endian encoding into `buf` would be `buf[0] = 232`, `buf[1] = 3`
  - big endian encoding into `buf` would be `buf[0] = 3`, `buf[1] = 232`

So, `BS-1` refers to `buf[BS-1]`, and in little endian this will hold the most significant byte, whereas in big endian this will hold the least significant byte.

I also think it's important to decouple "left to right"/"right to left" from "litte endian"/"big endian".

Does this clarify anything?

Reply:
okay, got it. @ed255 
Thanks for details!